contributors/design-proposals/daemonset-update.md

```diff
@@ -0,0 +1,222 @@
+# DaemonSet Updates
+
+**Author**: @madhusudancs, @lukaszo, @janetkuo
+
+**Status**: Proposal
+
+## Abstract
+
+A proposal for adding the update feature to `DaemonSet`. This feature will be
+implemented on server side (in `DaemonSet` API). 
+
+Users already can update a `DaemonSet` today (Kubernetes release 1.5), which will
+not cause changes to its subsequent pods, until those pods are killed. In this
+proposal, we plan to add a "RollingUpdate" strategy which allows DaemonSet to
+downstream its changes to pods. 
+
+## Requirements
+
+In this proposal, we design DaemonSet updates based on the following requirements:
+
+- Users can have a controlled number of in-flight updates of a DaemonSet (e.g.
+  DaemonSet only updates one pod at a time)
+- Users can monitor the status of a DaemonSet update (e.g. the number of pods
+  that are updated and healthy)
+- A broken DaemonSet update should not continue 
```

Comment:
That mechanism works fine. @erictune was questioning if we want to gate DS rollouts using the same mechanism we use in Deployments and I don't have a strong opinion since DS are destined to cover different use-cases from what Deployments cover.